### PR TITLE
Cleanup and exit if cannot download valid connection info.

### PR DESCRIPTION
**Pull request by vwbusguy**
https://github.com/rancher/rancher/issues/43991

This PR fixes an issue where if maximum retries are met attempting to download the rancher2_connection_info.json, the script continues with invalid data in that file causing the rancher-system-agent to fail to start.  This change cleans up the errant json file and exits the installer instead of continuing to setup a service doomed for failure.

Ref. SUSE Support case #01054437

### CODE REVIEW

**Comment by vwbusguy** [comment]
Addresses and possibly fixes #145

**Review by jakefhyde — CHANGES_REQUESTED** [review]
Thank you for opening this PR. Overall the proposed change seems like a positive to me, and I don't see any immediate downsides.

**Review by jakefhyde — APPROVED** [review]
Approved, pending review by @Oats87

**Review by Oats87 — APPROVED** [review]

**Pull request merged**
